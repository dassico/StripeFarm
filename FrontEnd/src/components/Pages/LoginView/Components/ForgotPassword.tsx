/* External imports */
import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {
    FormControl,
    InputLabel,
    Input,
    FormHelperText,
    Button,
    Typography
} from '@material-ui/core';

/* Local imports */
import Cloudia from '../../../../assets/Cloudia.png';

/*
    Styles for the view
*/
const useStyles = makeStyles(theme => ({
    // Style parameters for the header
    header: {
        width: '40%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'column',
        [theme.breakpoints.down('xs')]: {
            marginTop: 50
        }
    },
    // Height and with of the logo
    logo: {
        height: 200,
        width: 200
    },
    // Font for the description
    description: {
        fontSize: 26,
        fontWeight: 900
    },
    // Layout of the form
    form: {
        width: '40%',
        [theme.breakpoints.down('xs')]: {
            margin: '50px 0 0 0',
            width: '90%'
        }
    },
    // Margin on bottom of the form
    formControl: {
        marginBottom: theme.spacing(1)
    },
    // Style parameters for the textbutton
    textbutton: {
        padding: theme.spacing(1.5, 4),
        color: theme.palette.primary.dark,
        transition: 'all 0.2s ease',
        '&:hover': {
            color: theme.palette.primary.main
        }
    },
    // Style parameters submit button
    submit: {
        margin: theme.spacing(2, 0),
        padding: theme.spacing(1.5, 4),
        fontSize: 14,
        fontWeight: 900,
        color: 'white',
        background: theme.palette.primary.main,
        boxShadow:
            '0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06)',
        transition: 'all 0.2s ease',
        '&:hover': {
            background: theme.palette.primary.dark
        }
    },
    // Allignment of flex class
    flex: {
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center'
    }
}));

// Interface for parent function
interface ChildProps {
    cancel: () => void;
}

// Initialize email parameter to empty string
const initialState = {
    email: ''
};

/**
 * ForgotPasswordComponent() renders and returns the ForgotPassword dialog
 * @param props contains the cancel parameter from the parent class
 */
const ForgotPasswordComponent: React.FC<ChildProps> = (props: any) => {
    const classes = useStyles();

    /**
     * email: contains the email adres and is set to the initial state
     * error: contains the error messages based on the initial state
     */
    const [{ email }, setInfo] = useState(initialState);
    const [error, setError] = useState(initialState);

    /*
        onChange(): function that handles the inputs of the form
        @param e: event that triggered the onChange function
    */
    const onChange = (e: any) => {
        const { id, value } = e.target;
        setInfo(prevState => ({ ...prevState, [id]: value }));
    };

	/*
        handleSubmit(): handles the submit of the form
    */
    const handleSubmit = () => {
        // Reset errors
        setError(initialState);

        // Email field does not match regex
        if (!email.match('.+@.+..+')) {
            setError(error => ({
                ...error,
                email: 'Please use a valid email address'
            }));
            return;
        }

        // Request reset
        console.log(email);
    };

    return (
        <React.Fragment>
            {/* Header of the dialog, contains logo of stripefarmer */}
            <div className={classes.header}>
                <img className={classes.logo} src={Cloudia} alt="Cloudia" />
                <Typography className={classes.description} color="primary">
                    Reset Password
                </Typography>
            </div>
            {/* Form for email */}
            <form
                className={classes.form}
                noValidate
                // handleSubmit() if the Enter key is pressed
                onKeyPress={event => {
                    if (event.key === 'Enter') {
                        handleSubmit();
                    }
                }}
            >
                <Typography component="h5">
                    Enter your email address to reset your password.
                </Typography>
                <FormControl className={classes.formControl} required fullWidth>
                    {/* Input field for email */}
                    <InputLabel htmlFor="email">Email</InputLabel>
                    <Input
                        id="email"
                        type="email"
                        value={email}
                        onChange={onChange}
                        aria-describedby="email-error-text"
                    />
                    <FormHelperText id="email-error-text" error>
                        {error.email}
                    </FormHelperText>
                </FormControl>
                {/* Form cancel and submit buttons */}
                <div className={classes.flex}>
                    <Button
                        className={classes.textbutton}
                        onClick={props.cancel}
                    >
                        Cancel
                    </Button>
                    <Button className={classes.submit} onClick={handleSubmit}>
                        Reset password
                    </Button>
                </div>
            </form>
        </React.Fragment>
    );
};

export default ForgotPasswordComponent;
