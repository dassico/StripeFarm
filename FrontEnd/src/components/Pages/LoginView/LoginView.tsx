/* External imports */
import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';

/* Libraries */
import { Redirect } from 'react-router';

/* Local imports */
import Login from './Components/Login';
import ForgotPassword from './Components/ForgotPassword';
import SignUp from './Components/Signup';
import RegisterFarm from '../../Views/AddFarmView/AddFarm';
import { loggedIn } from '../../../utils/Controllers/UserController';

/*
    Styles for the view
*/
const useStyles = makeStyles(theme => ({
    root: {
        minHeight: '100vh',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        background:
            'linear-gradient(rgba(0,0,0,0.8), rgba(0,0,0,0.8)), url(https://images.pexels.com/photos/2132171/pexels-photo-2132171.jpeg)',
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat',
        // backgroundImage: "url(https://images.pexels.com/photos/2132171/pexels-photo-2132171.jpeg)" ,
        backgroundPosition: 'center',
        [theme.breakpoints.down('xs')]: {
            background: 'white'
        }
    },
    paper: {
        height: 480,
        width: 860,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-evenly',
        background: 'white',
        borderRadius: 5,
        boxShadow:
            '0 10px 30px -12px rgba(0, 0, 0, 0.42), 0 4px 25px 0px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
        [theme.breakpoints.down('xs')]: {
            height: '100vh',
            width: '80vw',
            boxShadow: 'none',
            justifyContent: 'flex-start',
            flexDirection: 'column'
        }
    }
}));

/*
    Renders the high-level login component
*/
const LoginView: React.FC = () => {
    const classes = useStyles();

    // Switch view to ForgotPassword
    const forgotPassword = () => {
        setView(<ForgotPassword cancel={cancel} />);
    };

    // Switch view to RegisterFarm
    const doFarmRegister = () => {
        setView(<RegisterFarm />);
    };

    // Switch view to Login
    const cancel = () => {
        setView(<Login forgotPassword={forgotPassword} signup={signup} />);
    };

    // Switch view to SignUp
    const signup = () => {
        setView(<SignUp cancel={cancel} doFarmRegister={doFarmRegister} />);
    };

    // Local high-level state
    const [view, setView] = React.useState(
        <Login forgotPassword={forgotPassword} signup={signup} />
    );

    // Check if logged in
    if (loggedIn()) {
        return <Redirect to="/" />;
    }

    /*
        Returns the grid with the login view
    */
    return (
        <Grid container component="main" className={classes.root}>
            <CssBaseline />
            <main className={classes.paper}>{view}</main>
        </Grid>
    );
};

export default LoginView;
