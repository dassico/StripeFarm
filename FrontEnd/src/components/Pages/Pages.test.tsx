/* This test file contains tests for the 404 page, login page, sign up page
 * and the forgot password page.
 */

import React from 'react';
import { create } from "react-test-renderer";
import Notfound from './404/404';
import { BrowserRouter as Router } from 'react-router-dom';
import LoginView from './LoginView/LoginView';
import SignUp from './LoginView/Components/Signup';
import ForgotPassword from './LoginView/Components/ForgotPassword';

const mockFunction = () => {
    return;
};
/**
 * This is a test case for the 404 page
 */
test('404 page renders and matches snapshot', () => {
    const view = create(<Router><Notfound /></Router>)
    expect(view.toJSON()).toMatchSnapshot();
});
/**
 * This is a test case for the login page
 */
test('login page renders with login view and matches snapshot', () => {
    const view = create(<LoginView />);
    expect(view.toJSON()).toMatchSnapshot();
});
/**
 * This is a test case for the signup view
 */
test('sign up view renders and matches snapshot', () => {
    const view = create(<SignUp cancel={mockFunction} doFarmRegister={mockFunction}/>);
    expect(view.toJSON()).toMatchSnapshot();
});
/**
 * This is a test case for the forgotpassword view
 */
test('Forgotpassword view renders and matches snapshot', () => {
    const view = create(<ForgotPassword cancel={mockFunction} />);
    expect(view.toJSON()).toMatchSnapshot();
});
