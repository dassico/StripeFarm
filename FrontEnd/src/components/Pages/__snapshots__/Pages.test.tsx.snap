// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page renders and matches snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <div
      className="MuiBox-root MuiBox-root-10"
    />
    <img
      alt="ImageCloudia"
      src="CloudiaDead.png"
      style={
        Object {
          "height": 250,
          "width": 250,
        }
      }
    />
    <h1
      className="MuiTypography-root MuiTypography-h1"
    >
      404
    </h1>
    <div
      className="MuiBox-root MuiBox-root-41"
    />
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Page not found
    </h5>
    <div
      className="MuiBox-root MuiBox-root-42"
    />
    <a
      href="/login"
      onClick={[Function]}
    >
      Back to login
    </a>
  </div>
</main>
`;

exports[`Forgotpassword view renders and matches snapshot 1`] = `
Array [
  <div
    className="makeStyles-header-268"
  >
    <img
      alt="Cloudia"
      className="makeStyles-logo-269"
      src="Cloudia.png"
    />
    <p
      className="MuiTypography-root makeStyles-description-270 MuiTypography-body1 MuiTypography-colorPrimary"
    >
      Reset Password
    </p>
  </div>,
  <form
    className="makeStyles-form-271"
    noValidate={true}
    onKeyPress={[Function]}
  >
    <h5
      className="MuiTypography-root MuiTypography-body1"
    >
      Enter your email address to reset your password.
    </h5>
    <div
      className="MuiFormControl-root makeStyles-formControl-272 MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="email"
      >
        Email
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="email-error-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-error Mui-required"
        id="email-error-text"
      >
        
      </p>
    </div>
    <div
      className="makeStyles-flex-275"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text makeStyles-textbutton-273"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Cancel
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text makeStyles-submit-274"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Reset password
        </span>
      </button>
    </div>
  </form>,
]
`;

exports[`login page renders with login view and matches snapshot 1`] = `
<main
  className="MuiGrid-root makeStyles-root-43 MuiGrid-container"
>
  <main
    className="makeStyles-paper-44"
  >
    <div
      className="makeStyles-header-148"
    >
      <img
        alt="Cloudia"
        className="makeStyles-logo-149"
        src="Cloudia.png"
      />
      <p
        className="MuiTypography-root makeStyles-description-150 MuiTypography-body1 MuiTypography-colorPrimary"
      >
        Welcome
      </p>
    </div>
    <form
      className="makeStyles-form-151"
      noValidate={true}
      onKeyPress={[Function]}
    >
      <div
        className="MuiFormControl-root makeStyles-formControl-152 MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="email"
        >
          Email
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="email-error-text"
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="email"
            value=""
          />
        </div>
        <p
          className="MuiFormHelperText-root Mui-error Mui-required"
          id="email-error-text"
        >
          
        </p>
      </div>
      <div
        className="MuiFormControl-root makeStyles-formControl-152 MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="password"
        >
          Password
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="password-error-text"
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
        <p
          className="MuiFormHelperText-root Mui-error Mui-required"
          id="password-error-text"
        >
          
        </p>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text makeStyles-submit-154 MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Log in
        </span>
      </button>
      <div
        className="makeStyles-flex-155"
      >
        Forgot password?
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text makeStyles-textbutton-153"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Reset
          </span>
        </button>
      </div>
      <div
        className="makeStyles-flex-155"
      >
        Don't have an account?
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text makeStyles-textbutton-153"
          disabled={false}
          id="signupButton"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Sign up
          </span>
        </button>
      </div>
    </form>
  </main>
</main>
`;

exports[`sign up view renders and matches snapshot 1`] = `
Array [
  <div
    className="makeStyles-header-250"
  >
    <img
      alt="Cloudia"
      className="makeStyles-logo-251"
      src="Cloudia.png"
    />
    <p
      className="MuiTypography-root makeStyles-description-252 MuiTypography-body1 MuiTypography-colorPrimary"
    >
      Create a new account
    </p>
  </div>,
  <form
    className="makeStyles-form-253"
    noValidate={true}
    onKeyPress={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-254 MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink={false}
            htmlFor="firstname"
          >
            First name
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-describedby="firstname-error-text"
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="firstname"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <p
            className="MuiFormHelperText-root Mui-error Mui-required"
            id="firstname-error-text"
          >
            
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-254 MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink={false}
            htmlFor="lastname"
          >
            Last name
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-describedby="lastname-error-text"
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="lastname"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <p
            className="MuiFormHelperText-root Mui-error Mui-required"
            id="lastname-error-text"
          >
            
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root makeStyles-formControl-254 MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="email"
      >
        Email
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="email-error-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-error Mui-required"
        id="email-error-text"
      >
        
      </p>
    </div>
    <div
      className="MuiFormControl-root makeStyles-formControl-254 MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="password"
      >
        Password
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="password-error-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-error Mui-required"
        id="password-error-text"
      >
        
      </p>
    </div>
    <div
      className="MuiFormControl-root makeStyles-formControl-254 MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="confirm"
      >
        Confirm password
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="confirm-error-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="confirm"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-error Mui-required"
        id="confirm-error-text"
      >
        
      </p>
    </div>
    <div
      className="makeStyles-flex-257"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text makeStyles-textbutton-255"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Cancel
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text makeStyles-submit-256"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sign up
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>,
]
`;
